### DISCUSSION TITLE
Joining Snapshots!

### DISCUSSION THREAD

**Discussion by lbenezra-FA** [dbt-labs/devhub-writing#90] [General]
**What's your key point?**
Properly joining snapshots is hard. Winnie and I figured out how to do it logically across many models. 

**Prior art:**
*Any other posts that exist on this topic (here or elsewhere).*
Not that I'm aware!

**Link to notes / draft:**
*Google docs preferred, please set sharing to `anyone with the link can view`.*
[Trials and Tribulations of Joining Snapshots
](https://docs.google.com/document/d/1xH82PG1d28Fv5jwCY9WENW-3ERxjNUMSK5LLwIvfYz0/edit?usp=sharing)
**Any open questions / requests for help from the group?**
Would love some thoughts around how to make this simpler / easier for people to understand.

**Comment by jasnonaz** [comment]
Hi @lbenezra-FA  - thank you so much for putting this truly wonderful post together. You are an _extremely_ strong technical writer and the visuals + code snippets bring this post to the next level.

The post right now has 98% of the building blocks in place - we just need to add in the some structural bits to make it all make sense. Right now the piece kind of loses the reader at the transition from the introduction to the step by step solution. I think we have a really clear path to making it much more intuitive for people that aren't already familiar with this solution.

First up - let's outline our solution before we go into the detailed section. Particularly where you write 

> 
Okay so we’ve ruled out the easy way. But we still need to build a joined table of history! So let’s tackle that O(n*m*a*b*c*d…*q) problem. 
> 
instead of diving straight into the step by step solutions there, we could give a one to two paragraph explanation of the solution so that people have context once we dive deeper. I would also suggest outlining the steps you'll be taking people through. An example of this structure can be found in [Joel's post on upgrading dbt projects](https://docs.getdbt.com/blog/upgrade-dbt-without-fear). Once we've done this, we're halfway there.

Next, can we go into each of the steps and summarize up top what the exact purpose of the step is and how it ties into the larger solution. This could be done via bullet points, but for the flow of this it feels like text explanations will work great. Just a little bit of extra context so people can understand what they're reading through.

Once we've done that the post will be looking 💯  💯

**Comment by lbenezra-FA** [comment]
thank you so much @jasnonaz ! This is incredibly helpful feedback. I'll be working on this today and will ping you when it's ready for another look. thank you for giving such thoughtful comments and suggestions!

**Comment by lbenezra-FA** [comment]
@jasnonaz Ready for another review! I added all of your suggestions. You made a very good point about a note being a separate mini-blog post, which I think totally makes sense. I also think the de-duping logic might work well as a separate blog post. in order for the snapshot joins to work, it's _crucial_ for the data to not have duplicates, but that's not always a step everyone will need to take. 
let me know what you think of these changes. Happy to discuss anything at any time!

**Comment by jasnonaz** [comment]
Hi @lbenezra-FA this is looking great! I think we have two options on how to proceed here. 

1. Publish this piece as is (with just a few open comments on sentence / wording level changes).
2. Split steps 1 and 2 out into a separate piece on removing dupes, publish that first, then publish this piece. I'd say a slight preference for that one as a post on removing dupes is useful across a huge number of use cases and would love to have it to refer to, but I want to be aware of time commitment / capacity on your end. Totally happy to pub the piece now and then we can add a general piece on removing duplicates to the backlog.

**Comment by lbenezra-FA** [comment]
@jasnonaz I like option 2! Happy to start a second blog post on dupes this week. It should be relatively short. I'll open another issue.

**Comment by jasnonaz** [comment]
Awesome!! Looking forward to this.

**Comment by lbenezra-FA** [comment]
Opened a [new issue](https://github.com/dbt-labs/dbt-technical-blog-writing/issues/78) with a first draft!

**Comment by gwenwindflower** [comment]
closed and transferred here https://github.com/dbt-labs/docs.getdbt.com/discussions/1143
